### ISSUE TITLE
IdNow crashes when back button pressed on Terms & Conditions screen

### ISSUE THREAD

**Issue by stramek** [idnow/de.idnow.android#185]
### Description
Hello! I'm trying to upgrade SDK version of IdNow in the application, but after upgrade I encountered an issue which crashes the application. I was able to reproduce it with sample IdNow project.

### Reproduce steps

1) Compile sample project https://github.com/idnow/de.idnow.android-sample with SDK between `6.1.5` and `6.3.0`
2) Provide test transaction token to the `IDnowSDK.setTransactionToken(String)` method
3) Launch application
4) Press `Video-identifizierung starten` or `Photo-Identifizierung starten` button
5) Screen with `Terms & Conditions` should appear
6) Press back button
7) SDK crash

It's possible, that the issue also appears for `6.4.0` but I'm unable to reach `Terms & Conditions` screen, because SDK crashes right away after pressing Video-identifizierung starten` or `Photo-Identifizierung starten` button. Created issue here: https://github.com/idnow/de.idnow.android/issues/184

### Reproduced on devices

- Nexus 5, Android 6.0.1
- Pixel 6, Android 13

I was testing only those two devices.

### Stacktrace

```
FATAL EXCEPTION: main
	Process: de.idnow, PID: 29119
	java.lang.NullPointerException: Attempt to invoke virtual method 'void de.idnow.insights.ModuleSessions$Sessions.endSession()' on a null object reference
		at de.idnow.sdk.Activities.Activities_VideoOverviewCheckActivity.onBackPressed(Activities_VideoOverviewCheckActivity.java:438)
		at android.app.Activity.onKeyUp(Activity.java:3920)
		at android.view.KeyEvent.dispatch(KeyEvent.java:2925)
		at android.app.Activity.dispatchKeyEvent(Activity.java:4263)
		at androidx.core.app.ComponentActivity.superDispatchKeyEvent(ComponentActivity.java:122)
		at androidx.core.view.KeyEventDispatcher.dispatchKeyEvent(KeyEventDispatcher.java:84)
		at androidx.core.app.ComponentActivity.dispatchKeyEvent(ComponentActivity.java:140)
		at androidx.appcompat.app.AppCompatActivity.dispatchKeyEvent(AppCompatActivity.java:599)
		at androidx.appcompat.view.WindowCallbackWrapper.dispatchKeyEvent(WindowCallbackWrapper.java:59)
		at androidx.appcompat.app.AppCompatDelegateImpl$AppCompatWindowCallback.dispatchKeyEvent(AppCompatDelegateImpl.java:3090)
		at com.android.internal.policy.DecorView.dispatchKeyEvent(DecorView.java:402)
		at android.view.ViewRootImpl$ViewPostImeInputStage.processKeyEvent(ViewRootImpl.java:6687)
		at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:6553)
		at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:6013)
		at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:6070)
		at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:6036)
		at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:6201)
		at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:6044)
		at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:6258)
		at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:6017)
		at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:6070)
		at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:6036)
		at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:6044)
		at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:6017)
		at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:6070)
		at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:6036)
		at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:6234)
		at android.view.ViewRootImpl$ImeInputStage.onFinishedInputEvent(ViewRootImpl.java:6414)
		at android.view.inputmethod.InputMethodManager$PendingEvent.run(InputMethodManager.java:3649)
		at android.view.inputmethod.InputMethodManager.invokeFinishedInputEventCallback(InputMethodManager.java:3169)
		at android.view.inputmethod.InputMethodManager.finishedInputEvent(InputMethodManager.java:3160)
		at android.view.inputmethod.InputMethodManager$ImeInputEventSender.onInputEventFinished(InputMethodManager.java:3626)
		at android.view.InputEventSender.dispatchInputEventFinished(InputEventSender.java:154)
		at android.os.MessageQueue.nativePollOnce(Native Method)
		at android.os.MessageQueue.next(MessageQueue.java:335)
		at android.os.Looper.loopOnce(Looper.java:161)
		at android.os.Looper.loop(Looper.java:288)
		at android.app.ActivityThread.main(ActivityThread.java:7884)
		at java.lang.reflect.Method.invoke(Native Method)
		at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
		at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
```

**Comment by erayasan** [comment]
Hi, according to my experiences this issue is not specific to version 6.4.0.  It is  reproducible for each 6.x.x versions of this sdk. Even I was able to completely able to finish identification process when navigation back from sdk activity to apps activity this sdk crashes the application.

**Comment by onursahindur** [comment]
This issue also exists on our platform too.
When we complete IDNow process in test scenario we see a success page. After hit okey button it not completely remove from screen, it goes back to terms and conditions page. After we hit back button, the SDK causes a crash.

**Comment by stramek** [comment]
Fixed in 6.6.0
